autogpt_platform/autogpt_libs/autogpt_libs/rate_limit/middleware.py

```diff
@@ -0,0 +1,31 @@
+from fastapi import Request, HTTPException
+from .limiter import RateLimiter
+
+
+async def rate_limit_middleware(request: Request, call_next):
+    """FastAPI middleware for rate limiting API requests."""
+    limiter = RateLimiter()
+
+    if not request.url.path.startswith("/api"):
+        return await call_next(request)
+
+    api_key = request.headers.get("Authorization")
+    if not api_key:
+        return await call_next(request)
```

Comment:
Can we limit non-auth requests like by IP ? or is this not the point of the PR

Reply:
I like this idea, it's not in scope for this PR but will add to a ticket